akka-persistence/src/main/scala/akka/persistence/Persistence.scala

```diff
@@ -215,6 +253,34 @@ class Persistence(val system: ExtendedActorSystem) extends Extension {
     system.systemActorOf(pluginActorProps, pluginActorName)
   }
 
+  private def createAdapter(configPath: String): Option[ModelAdapter[Any]] = {
+    val pluginConfig = system.settings.config.getConfig(configPath)
+    if (!pluginConfig.hasPath("adapter")) None
+    else {
+      val adapterClassName = pluginConfig.getString("adapter")
```

Comment:
Oh - no I think you got me right, see here for tagger impl: https://github.com/ktoso/akka/commit/410a4a8738296187076d1b74074a6ee15798c8a4#diff-aa1e901f1dd62bc57d4b969106f64c27R70 It's as serializers "pick most specific one" and allowed for multiple ones to be combined into one (that part we do not need for adapters).

I somehow thought "one adapter per journal", but now that you mention it "serializer like registration" for those makes much sense, so I'll bring back that idea. :)